### ISSUE TITLE
Dock: animation behaviour fault when scrolling page

### ISSUE THREAD

**Issue by geziefer** [primefaces/primefaces#1831]
## 1) Environment
- PrimeFaces version: 5.3
- Does it work on the newest PrimeFaces version? no
- Application server + version: Tomcat 7
- Affected browsers: recent versions of Chrome, Firefox, Edge, IE
## 2) Expected behavior

When using a dock menu positioned on the bottom, it is expected that it behaves the same regardless being on a short page which fits into browser window or on a long page which is scrollable.
## 3) Actual behavior

When scrolling down, the animation which magnifies the icons does not work on mouse-over, but only if moving the mouse a bit above the icon. The more the page is scrolled down, the more space between icon and mouse pointer is needed to trigger the animation.
## 4) Steps to reproduce
- Go to PrimeFaces ShowCase to "Menu - Dock" page (http://www.primefaces.org/showcase/ui/menu/dock.xhtml)
- Make the browser window very small, so that only the dock menu and a bit of text above and a larger scrollbar is visible
- Scroll down a good part of the page
- Try to do mouse-over the icons and see that they do nothing
- Try the same a bit above the icons and see the animation
## 5) Sample XHTML

[See showcase example dock.xhtml]

```
<p:dock position="bottom">
    <p:menuitem value="Home" icon="/resources/demo/images/dock/home.png" url="#"/>
    <p:menuitem value="Music" icon="/resources/demo/images/dock/music.png" url="#"/>
</p:dock>

```
## 6) Sample bean

Not needed

I posted this as a question at StackOverflow (http://stackoverflow.com/questions/39991422/primefaces-dock-menu-animation-failure-when-scrolling) but after checking today that your showcase behaves the same with the recent version 6.0.6, I  filed it as a bug.

**Comment by geziefer** [comment]
It's not only a non-working animation which makes the whole menu look wretched, but it also prevents other buttons from being pressed which are beyond the point where the menu thinks it has to show the bubble and let the menu item be clicked - which is a pretty major thing I'd say and makes it unusable.

Will someone care about that or do we have to abandon the useless dock menu?

**Comment by tandraschko** [comment]
> Will someone care about that

You have 3 solutions:
- you can use PrimeFaces support to get the bug fixed
- wait until someone fixes it
- analyze the problem by yourself and provide a PR :)

**Comment by geziefer** [comment]
Hehe, I knew I'll get immediate comments when conplaining about the bug - but it was meant as a question, because I had the fear that no one at all might care about it, and I regard it as a major bug in the component, so I'd like to know if help is on the horizon or if I have to ommit the menu and do something differently.

**Comment by CodeGolfWhatsapp** [comment]
https://www.youtube.com/watch?v=J00yDkAkViA&feature=gp-n-y&google_comment_id=z12jdt5w4s2kevgeb23zzzsygwrsvzdin04

**Comment by melloware** [comment]
Fixed with this PR:  https://github.com/primefaces/primefaces/pull/2997

**Comment by geziefer** [comment]
Is it part of version 6.2.1? (label is not set, but I tried it out and it seemed to be fixed).

However, even though the main animation bug is fixed, it is not quite ok yet:
If you start with a half-sized window, then scroll down the page, the behaviour is correct.
But when you maximize the browser window again, the animation is wrong and starts in about the middle of the screen instead of the bottom where the dock menu is.
Similar problems occur when playing around with the window size - the animation should always occur when the mouse approaches the icons, but it often starts much earlier, sometimes even on top of the page.

Reproducable with both a recent Firefox and Chrome, on Windows 10, Primefaces 6.2.1 in Java EE 7 application on WildFly 11.

**Comment by melloware** [comment]
Yes it was fixed for PF 6.2 so its in 6.2.1.     And not its not perfect it was fixed based on this blog post.

http://www.monkey-business.biz/181/jquery-interfaces-fisheye-position-fixed-patch/#jquery-interfaces-fisheye-position-fixed-patch_english
